community/modules/scheduler/htcondor-access-point/main.tf

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, { ghpc_module = "htcondor-access-point" })
+}
+
+locals {
+  network_storage_metadata = var.network_storage == null ? {} : { network_storage = jsonencode(var.network_storage) }
+  oslogin_api_values = {
+    "DISABLE" = "FALSE"
+    "ENABLE"  = "TRUE"
+  }
+  enable_oslogin_metadata = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+  metadata                = merge(local.network_storage_metadata, local.enable_oslogin_metadata, var.metadata)
+
+  name_prefix = "${var.deployment_name}-ap"
+
+  example_runner = {
+    type        = "data"
+    destination = "/var/tmp/helloworld.sub"
+    content     = <<-EOT
+      universe       = vanilla
+      executable     = /bin/sleep
+      arguments      = 1000
+      output         = out.\$(ClusterId).\$(ProcId)
+      error          = err.\$(ClusterId).\$(ProcId)
+      log            = log.\$(ClusterId).\$(ProcId)
+      request_cpus   = 1
+      request_memory = 100MB
+      # if unset, defaults to false
+      +RequireSpot   = true
+      queue
+    EOT
+  }
+
+  all_runners = concat(
+    var.access_point_runner,
+    [local.schedd_runner],
+    var.autoscaler_runner,
+    [local.example_runner]
+  )
+  disk_size_gb = max(var.disk_size_gb, data.google_compute_image.htcondor.disk_size_gb)
+
+  ap_config = templatefile("${path.module}/templates/condor_config.tftpl", {
+    htcondor_role       = "get_htcondor_submit",
+    central_manager_ips = var.central_manager_ips
+    spool_dir           = "${var.spool_parent_dir}/spool",
+  })
+
+  ap_object = "gs://${var.htcondor_bucket_name}/${google_storage_bucket_object.ap_config.output_name}"
+  schedd_runner = {
+    "type"        = "ansible-local"
```

Comment:
Is there a reason for quoting obj keys? `"type" -> type`

Reply:
Developer error. Addressed in update.